### PR TITLE
Detect SIMD instruction set.

### PR DESCRIPTION
**Pull request by fabianbl**
Updated cmake script for detection of the available SIMD instruction set.

### CODE REVIEW

File: cmake/DetectSIMD.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Detect the preprocessor directives which are set by the compiler.
+execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dM -E -x c /dev/null
+                OUTPUT_VARIABLE PREPROCESSOR_DIRECTIVES)
+
+if (PREPROCESSOR_DIRECTIVES MATCHES "__SSE2__")
```
Comment (mfehr) [inline cmake/DetectSIMD.cmake]: same as in other PRs, is this covering SSE3?

**Comment by mfehr** [comment]
test this please

**Comment by mfehr** [comment]
ok to test

**Comment by mfehr** [comment]
add to whitelist

**Comment by mfehr** [comment]
@fabianbl I added a jenkins job for this, we didn't have one before... Looks like we broke quite a few things with our adaptations. We will need to fix this, but for now just quickly check if the build job fails on anything that could relate to your change and then we can just merge it.

FYI: @schneith @dymczykm

**Comment by mfehr** [comment]
test this please

**Comment by mfehr** [comment]
ok, it looks like only osx is broken, so that's not so bad.

**Comment by mfehr** [comment]
test this please (disabled osx for now)

**Comment by fabianbl** [comment]
test this please

**Pull request merged**